directory/datadog_checks/directory/directory.py

```diff
@@ -46,12 +51,21 @@ def check(self, _):
                 "Either directory '{}' doesn't exist or the Agent doesn't "
                 "have permissions to access it, skipping.".format(self.config.abs_directory)
             )
-
-            if not self.config.ignore_missing:
-                raise ConfigurationError(msg)
+            # report missing directory and skip check
+            if self.config.report_missing:
+                self.log.info('service check SERVICE_DIRECTORY_EXISTS')
+                self.service_check(name=SERVICE_DIRECTORY_EXISTS, status=self.WARNING, message=msg)
```

Comment:
To avoid the service check to go into the "no data" state you should also submit it with OK value if the directory exists.